lib/chromedriver.js

```diff
@@ -525,6 +536,37 @@ class Chromedriver extends events.EventEmitter {
     this.capabilities = toW3cCapNames(this.capabilities);
   }
 
+  /**
+   * Sync the protocol by reading the given output
+   *
+   * @param {string} out The output of WebDriver process start
+   */
+  detectWebDriverProtocol(out) {
+    if (this.driverVersion) {
+      // Nothing is done if the protocol was already detected
+      return;
+    }
+
+    // also print chromedriver version to logs
+    // will output something like
+    //  Starting ChromeDriver 2.33.506106 (8a06c39c4582fbfbab6966dbb1c38a9173bfb1a2) on port 9515
+    // Or MSEdge:
+    //  Starting Microsoft Edge WebDriver 111.0.1661.41 (57be51b50d1be232a9e8186a10017d9e06b1fd16) on port 9515
+    const match = WEBDRIVER_VERSION_PATTERN.exec(out);
+    if (match && match.length === 3) {
+      this.log.debug(`${match[1]} version: '${match[2]}'`);
+      this.driverVersion = match[2];
+      try {
+        this.syncProtocol();
+      } catch (e) {
+        this.log.warn(`Failed to sync the protocol as '${e}'`);
```

Comment:
I would rather simply rethrow here. There is no need to silence exceptions

Reply:
It seems like the entire appium process stops without closing the session. Maybe should change other places as well to raise an error with closing the session properly then. I'll take a look as a followup pr

```
[debug] [Chromedriver@dcf6] Microsoft Edge WebDriver version: '112.0.1722.7'
[Chromedriver@dcf6] Failed to sync the protocol as 'Error: error'
uncaughtException: Failed to sync the protocol as 'Error: error'
Error: Failed to sync the protocol as 'Error: error'
    at Object.wrappedLogger.errorAndThrow (/Users/kazu/GitHub/appium-chromium-driver/node_modules/appium-chromedriver/node_modules/@appium/support/lib/logging.js:115:34)
    at Chromedriver.detectWebDriverProtocol (/Users/kazu/GitHub/appium-chromium-driver/node_modules/appium-chromedriver/lib/chromedriver.js:564:18)
    at SubProcess.<anonymous> (/Users/kazu/GitHub/appium-chromium-driver/node_modules/appium-chromedriver/lib/chromedriver.js:635:14)
    at SubProcess.emit (node:events:527:28)
    at handleOutput (/Users/kazu/GitHub/appium-chromium-driver/node_modules/appium-chromedriver/node_modules/teen_process/lib/subprocess.js:172:14)
    at Socket.<anonymous> (/Users/kazu/GitHub/appium-chromium-driver/node_modules/appium-chromedriver/node_modules/teen_process/lib/subprocess.js:215:48)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:285:11)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
[debug] [Chromedriver@dcf6] Matched '/status' to command name 'getStatus'
[debug] [Chromedriver@dcf6] Proxying [GET /status] to [GET http://127.0.0.1:9515/status] with no body
[debug] [Chromedriver@dcf6] Got response with status 200: {"value":{"build":{"version":"112.0.1722.7 (ed8d40790a44198d5db9a9cdebbf59c6c4ab0dd0)"},"message":"msedgedriver ready for new sessions.","os":{"arch":"x86_64","name":"Mac OS X","version":"10.15.7"},"ready":true}}
[Chromedriver@dcf6] Starting W3C Chromedriver session with capabilities: {
[Chromedriver@dcf6]   "capabilities": {
[Chromedriver@dcf6]     "alwaysMatch": {
[Chromedriver@dcf6]       "ms:edgeOptions": {},
[Chromedriver@dcf6]       "goog:loggingPrefs": {
[Chromedriver@dcf6]         "browser": "ALL"
[Chromedriver@dcf6]       }
[Chromedriver@dcf6]     }
[Chromedriver@dcf6]   }
[Chromedriver@dcf6] }
[debug] [Chromedriver@dcf6] Matched '/session' to command name 'createSession'
[debug] [Chromedriver@dcf6] Proxying [POST /session] to [POST http://127.0.0.1:9515/session] with body: {"capabilities":{"alwaysMatch":{"ms:edgeOptions":{},"goog:loggingPrefs":{"browser":"ALL"}}}}
kazu$
```

Reply:
Changed to call `this.stop();` to stop the driver process. Then, in case the driver already started a browser process, the driver process stops it as well. the `stop` also prevent appium process stops in the above my comment's case. Should be good to not process the new session request since this case may the driver has weird status.